VideoLocker/src/main/java/org/edx/mobile/util/WeakList.java

```diff
@@ -16,12 +16,24 @@ public WeakList() {
         items = new ArrayList();
     }
 
-    public void add(int index, T element) {
-        items.add(index, new WeakReference(element));
+    public void clear(){
+        items.clear();
     }
 
-    public void add(T element) {
-        items.add(new WeakReference(element));
+    public WeakReference<T> add(int index, T element) {
+        synchronized (items) {
```

Comment:
If this list is accessed asynchronously, could index no longer be valid by the time we get the lock?